website/cue/reference/components/sources/internal_metrics.cue

```diff
@@ -552,6 +552,42 @@ components: sources: internal_metrics: {
 				mode: _mode
 			}
 		}
+		component_received_events_count: {
+			description: """
+				A histogram of Vector's internal batch sizes passed around in Vector's internal topology. Note that this
+				is different than sink-level batching. It is mostly useful for debugging performance issues in Vector
+				due to low internal batching.
```

Comment:
Could we make it clearer that it's the number of events in the batch, "size" generally becomes a point of confusion.

Reply:
👍 good call.